src/Core/View.js

```diff
@@ -141,6 +141,9 @@ function _preprocessLayer(view, layer, provider, parentLayer) {
         };
     }
 
+    layer.getCoords = layer.getCoords || (node => [node.extent]);
```

Comment:
I support the effort of moving things out of TileMesh but at the same time I don't think that adding a `getZoom` API for layers is a good idea.
The `zoom` (or level) concept only applies to parts of iTowns (tiled geometries and protocols) and making a global API out of a local concept should be avoided.

Reply:
FWIW, I have a wip branch that removes the notion of level from everyplaces where it doesn't make sense.

You can see the diff the branch here: https://github.com/iTowns/itowns/compare/master...peppsac:colorisation

The changeset is quite large so I'll have to split it in several PR / commits, but here's an overview:
  - splits LayeredMaterialNodeProcessing in 2 (color and texture)
  - moves code that depends on the protocol in the provider (see the new function: `canTextureBeImproved`) and simplifies a lot the texture processing code
  - allows coloring pointclouds with WMS textures (WMTS could work later, but it needs a bit more work) - see the pointcloud_lyon example)

Reply:
Do you agree on `getCoords` ?

I don't like `getZoom` altogether as well :). it is only used by the LayeredMaterial AFAIK and I have a huge PR on a refacto of this material that does not need getZoom anymore. So the long-term plan is to discard it altogether.

Reply:
> The zoom (or level) 

Is it the same thing ? why do we have both terminologies ?

Reply:
I think the `getCoords` function would be unnecessary in my branch as well because the only thing that prevents the removal in your branch is that LayeredMaterialNodeProcessing uses it.